KubeArmor/feeder/feeder.go

```diff
@@ -322,6 +322,25 @@ func NewFeeder(node *tp.Node) *Feeder {
 	}
 	fd.Listener = listener
 
+	if cfg.GlobalCfg.GRPC == "0" {
+		pidFile, err := os.Create(cfg.PIDFilePath)
+		if err != nil {
+			kg.Errf("Failed to create file %s", cfg.PIDFilePath)
+			return nil
+		}
+
+		defer pidFile.Close()
```

Comment:
It seems defer is not a preferred method to close the file 😅 We can perhaps use
```
defer func() {
  err := pidFile.Close()
  // ...
}()
```
Ref https://github.com/securego/gosec/issues/512#issuecomment-675286833

Reply:
We need to error check inside the defer func.